templates/common/_base/files/wait-for-ipsec-connect.yaml

```diff
@@ -9,25 +9,39 @@ contents:
       exit 0
     fi
 
-    # Modify existing IPsec connection entries with "auto=start"
-    # option and restart ipsec systemd service. This helps to
-    # establish IKE SAs for the existing IPsec connections with
-    # peer nodes. This option will be deleted from connections
-    # once ovs-monitor-ipsec process spinned up on the node by
-    # ovn-ipsec-host pod, but still it won't reestablish IKE SAs
-    # again with peer nodes, so it shouldn't be a problem.
-    if ! grep -q "auto=start" /etc/ipsec.d/openshift.conf; then
-      sed -i '/^.*conn ovn.*$/a\    auto=start' /etc/ipsec.d/openshift.conf
+    ovsipsecsvc="openvswitch-ipsec.service"
+    if ! systemctl is-active --quiet $ovsipsecsvc; then
+      echo "$ovsipsecsvc is not running, so make pluto to establish IKE SAs"
+
+      # Modify existing IPsec connection entries with "auto=start"
+      # option and restart ipsec systemd service. This helps to
+      # establish IKE SAs for the existing IPsec connections with
+      # peer nodes. This option will be deleted from connections
+      # once ovs-monitor-ipsec process spinned up on the node by
+      # ovn-ipsec-host pod, but still it won't reestablish IKE SAs
+      # again with peer nodes, so it shouldn't be a problem.
+      if ! grep -q "auto=start" /etc/ipsec.d/openshift.conf; then
+        sed -i '/^.*conn ovn.*$/a\    auto=start' /etc/ipsec.d/openshift.conf
+      fi
+      chroot /proc/1/root ipsec restart
     fi
-    chroot /proc/1/root ipsec restart
 
     # Wait for upto 60s to get IPsec SAs to establish with peer nodes.
     timeout=60
     elapsed=0
     desiredconn=""
     establishedsa=""
     while [[ $elapsed -lt $timeout ]]; do
-      desiredconn=$(grep -E '^\s*conn\s+' /etc/ipsec.d/openshift.conf | grep -v '%default' | awk '{print $2}' | tr ' ' '\n' | sort | tr '\n' ' ')
+      if systemctl is-active --quiet $ovsipsecsvc; then
+        # When openvswitch-ipsec.service is active (i.e. IPsec is enabled for
+        # east west traffic), the service may be just started before
+        # wait-for-ipsec-connect.service, it may take a while to populate
+        # /etc/ipsec.d/openshift.conf with IPsec connection entries.
+        # so retrieve connection names from ovs.
+        desiredconn=$(ovs-vsctl --columns=name find interface type=geneve | awk -F': ' '/name/ {print $2}' | sort | awk '{print $0"-in-1", $0"-out-1"}' | tr '\n' ' ')
+      else
+        desiredconn=$(grep -E '^\s*conn\s+' /etc/ipsec.d/openshift.conf | grep -v '%default' | awk '{print $2}' | tr ' ' '\n' | sort | tr '\n' ' ')
+      fi
```

Comment:
The ovs-vsctl method should work in both cases, so maybe just always use it instead of parsing openshift.conf?

Reply:
If ipsec is disabled in OVN, then those tunnels should not have ipsec configuration.   Can we filter by `remote_cert` being non-empty, for example?  Something like:
```
ovs-vsctl --columns=name find interface type=geneve options:remote_cert!='""'
```
?

Reply:
yes @igsilya, filtering tunnels on `remote_name` would handle both cases, thanks ! updated the script.